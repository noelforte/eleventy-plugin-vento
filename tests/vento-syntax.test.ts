import { EleventyTest } from './_eleventy-test-instance.js';
import { test } from 'vitest';

const testInstance = new EleventyTest('./tests/stubs-vento-syntax/', {
	pluginOptions: {
		autotrim: true,
	},
});

await testInstance.rebuild();

const matrix: [string, string][] = [
	['Can set/print variables', 'set'],
	['Can run functions', 'function'],
	['Can iterate through loops', 'for'],
	['Can print exported values', 'export'],
	['Can print imported values', 'import'],
];

test.for(matrix)('%s in templates', async ([_label, slug], { expect }) => {
	const result = testInstance.getBuildResultForUrl(`/${slug}/`);
	expect(result?.content).toMatchSnapshot();
});
