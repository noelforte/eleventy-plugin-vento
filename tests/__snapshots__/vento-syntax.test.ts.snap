// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Can iterate through loops in templates 1`] = `
"Eleventy keys:
version
generator
env
directories

Page keys:
inputPath
fileSlug
filePathStem
outputFileExtension
templateSyntax
date
rawInput
url
outputPath
"
`;

exports[`Can print exported values in templates 1`] = `
"
Release the possums!



	Release 5 million possums!



"
`;

exports[`Can print imported values in templates 1`] = `
"
Release the possums!
	Release 20 million possums!

"
`;

exports[`Can run functions in templates 1`] = `
"

	Hello, world!


	Release the possums!

"
`;

exports[`Can set/print variables in templates 1`] = `
"
my val
"
`;
